src/libraries/System.IO.MemoryMappedFiles/src/System/IO/MemoryMappedFiles/MemoryMappedFile.Unix.cs

```diff
@@ -190,7 +190,14 @@ private static SafeFileHandle CreateSharedBackingObject(Interop.Sys.MemoryMapped
             do
             {
                 mapName = GenerateMapName();
-                fd = Interop.Sys.ShmOpen(mapName, flags, (int)perms); // Create the shared memory object.
+                if (Interop.Sys.MemfdSupported)
+                {
+                    fd = Interop.Sys.MemfdCreate(mapName);
```

Comment:
If the information in flags and perms isn't factored in here, where does it get incorporated?

Reply:
`memfd_create` flags do not have direct equivalents for read-only or read-write permissions. The flags used with `memfd_create` are mainly related to file descriptor behavior (e.g., closing on exec and allowing sealing), not the memory protection levels. Therefore, it makes sense to keep `MFD_CLOEXEC` hardcoded in C.

It was missing `mmap` call to set the protection, which I have just added. Inheritance is set the same way as with `shm_open` (default: CLOEXEC, clear flag if Inheritable is requested from line 244).

Reply:
If flags and perms aren't relevant to the if block, should they be moved to the else block? They're only ever used there. I realize it's inside of a retry loop, but we expect retries to be rare bordering on non-existent.

Reply:
That said, would there be any hardening benefits to using seals as a stand-in for what perms was being used for?

Reply:
When using `shm_open` with read-only permissions (e.g., `O_RDONLY`) and mapping it with `mmap` with read-only protections (e.g., `PROT_READ`), the resulting memory mapping will not allow writing through that specific file descriptor and mapping. However, if another process has opened the same shared memory object with read-write permissions (e.g., `O_RDWR`), it can still write to the shared memory, and those changes will be visible to the read-only mappings.

With `memfd_create` there is no protection on fd by default. We can `write(fd)` unless we implement write sealing: https://github.com/am11/runtime/commit/38eca38a1f1240cbecb5f499477683460fec675c. This will make it readonly for current process (same as `shm_open`) as well as other processes (different than `shm_open`). While it is not exactly the drop-in replacement, but I think it is a goodness that we will be more hardened than `shm_open`?

Reply:
Yeah, other than the extra syscall, there doesn't appear to be a downside to setting seals and it will help to harden the permissions. I suggest we add it in.